pkg/stub/handler.go

```diff
@@ -368,44 +443,62 @@ func (h *Handler) CreateDefaultResourceIfNeeded() error {
 		if err != nil {
 			return h.processError(srcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "issues waiting for delete to complete: %v")
 		}
-		h.samplesResource = nil
-	} else {
-		srcfg = h.samplesResource
 	}
-	if srcfg != nil {
-		logrus.Println("SampleResource already received, not creating default")
-		return nil
-	}
-	// "4a" in the "startup" workflow, just create default
-	// resource and set up that way
-	srcfg = &v1alpha1.SamplesResource{}
-	srcfg.Name = v1alpha1.SamplesResourceName
-	srcfg.Kind = "SamplesResource"
-	srcfg.APIVersion = "samplesoperator.config.openshift.io/v1alpha1"
-	srcfg.Spec.Architectures = append(srcfg.Spec.Architectures, v1alpha1.X86Architecture)
-	srcfg.Spec.InstallType = v1alpha1.CentosSamplesDistribution
+
 	logrus.Println("creating default SamplesResource")
 	err = h.sdkwrapper.Create(srcfg)
 	if err != nil {
 		if !kerrors.IsAlreadyExists(err) {
 			return err
 		}
-		logrus.Println("got already exists error on create default, just going to forgo this time vs. retry")
-		// just return the raw error and initiate the sdk's requeue/ratelimiter setup
+		logrus.Println("got already exists error on create default")
 	}
 	h.deleteInProgress = false
-	return err
+	return nil
 }
 
-func (h *Handler) manageDockerCfgSecret(deleted bool, samplesResource *v1alpha1.SamplesResource, secret *corev1.Secret) error {
-	if secret.Name != v1alpha1.SamplesRegistryCredentials {
+func (h *Handler) WaitingForCredential(srcfg *v1alpha1.SamplesResource) error {
+	if srcfg.ConditionTrue(v1alpha1.ImportCredentialsExist) {
 		return nil
 	}
 
-	if h.samplesResource == nil && h.waitingForCredential == nil {
-		if !deleted {
-			h.registrySecret = secret
+	secret, err := h.secretclientwrapper.Get(h.namespace, v1alpha1.SamplesRegistryCredentials)
+	// have to check both since get will return an empty secret struct along with not found errs
+	// handle case where secret is created in sample operator namespace before SamplesResource
+	if secret != nil && !kerrors.IsNotFound(err) {
+		return h.manageDockerCfgSecret(false, srcfg, secret)
+	}
```

Comment:
per my other comment, I don't think you need to do this (looking for the secret manually).

I think all this method has to do is:

1) if credentials exist==true, return false(we aren't waiting for credentials)
2) if credentials exist == false && we need credentials(the checking you do below), return true (we are waiting for credentials)
3) else return false

Reply:
Might need a face to face conversation on this, but I'll try to capture here first.

- if secret event comes *AFTER* initial samplesresource event, then agreed, do not need the secret lookup nor call to `manageDockerCfgSecret` ... that will happen on the secret event processing

- if secret event comes *BEFORE* samplesresource event, then yeah agree but was uncomfortable because
 -- if the secret relists occur long enough, as we return errors in the secret path, then presumably on of the relists should have the samplesresource available and complete processing.
 -- at least with the SDK version I'm currently using, here is what is documented for relist in informer-sync.go:
```
	// Copy from deployment_controller.go:
	// maxRetries is the number of times a Vault will be retried before it is dropped out of the queue.
	// With the current rate-limiter in use (5ms*2^(maxRetries-1)) the following numbers represent the times
	// a Vault is going to be requeued:
	//
	// 5ms, 10ms, 20ms, 40ms, 80ms, 160ms, 320ms, 640ms, 1.3s, 2.6s, 5.1s, 10.2s, 20.4s, 41s, 82s
	maxRetries = 15

```

-- seems like it should be sufficient window ... but I'm still nervous about the underlying stuff changing on us ... and granted we may move to different underlying stuff anyway

I'll remove, but wanted to at least verbalize the rationale behind the presumably unneeded redundancy.